src/Composer/Repository/ComposerRepository.php

```
@@ -304,15 +298,7 @@ public function whatProvides(Pool $pool, $name)
             $url = str_replace(array('%package%', '%hash%'), array($name, $hash), $this->providersUrl);
             $cacheKey = 'provider-'.strtr($name, '/', '$').'.json';
         } else {
-            // BC handling for old providers-includes
-            $url = 'p/'.$name.'.json';
-
-            // package does not exist in this repo
-            if (!isset($this->providerListing[$url])) {
-                return array();
-            }
-            $hash = $this->providerListing[$url]['sha256'];
-            $cacheKey = null;
+            return array();
```

All repository parsing code appears to take the stance that unsupported properties are silently ignored, and repository data is extracted from properties it does know how to parse. Therefore I opted to revert these to the situation as if that property had never existed.

This specific code was BC'd in March 2013 I believe, so I think that's the best approach for this.